src/net/atlas/db.rs

```diff
@@ -0,0 +1,379 @@
+use rusqlite::types::ToSql;
+use rusqlite::Row;
+use rusqlite::Transaction;
+use rusqlite::{Connection, OpenFlags, NO_PARAMS};
+
+use std::collections::HashSet;
+use std::convert::From;
+use std::convert::TryFrom;
+use std::fs;
+
+use util::db::tx_begin_immediate;
+use util::db::DBConn;
+use util::db::Error as db_error;
+use util::db::{query_count, query_int, query_row, query_rows, u64_to_sql, FromColumn, FromRow};
+
+use util;
+use util::hash::{bin_bytes, hex_bytes, to_bin, to_hex, Hash160};
+use util::log;
+use util::macros::is_big_endian;
+use util::secp256k1::Secp256k1PrivateKey;
+use util::secp256k1::Secp256k1PublicKey;
+
+use vm::types::QualifiedContractIdentifier;
+
+use chainstate::burn::{BlockHeaderHash, ConsensusHash};
+use chainstate::stacks::StacksBlockId;
+use net::StacksMessageCodec;
+
+use super::{Attachment, AttachmentInstance};
+
+pub const ATLASDB_VERSION: &'static str = "23.0.0.0";
+
+const ATLASDB_SETUP: &'static [&'static str] = &[
+    r#"
+    CREATE TABLE attachments(
+        hash TEXT UNIQUE PRIMARY KEY,
+        content BLOB NOT NULL
+    );"#,
+    r#"
+    CREATE TABLE attachment_instances(
+        id INTEGER PRIMARY KEY AUTOINCREMENT,
+        content_hash TEXT,
+        created_at INTEGER NOT NULL,
+        consensus_hash STRING NOT NULL,
+        block_header_hash STRING NOT NULL,
+        block_id STRING NOT NULL,
```

Comment:
What's a `block_id`?

Reply:
Ah, this is a `StacksBlockId`.  We've been using `index_block_hash` as a database row name for this field.  Can you also use it here?

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/2013/commits/99324aff0cba86129cc109a0fc9aea01f45a7ecd